src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs

```diff
@@ -82,6 +82,74 @@ public OpenConnectionMessage(string connectionId, Claim[] claims, IDictionary<st
         public string Protocol { get; set; }
     }
 
+    /// <summary>
+    /// Close connections in a hub
+    /// </summary>
+    public class CloseConnectionsWithAckMessage : ExtensibleServiceMessage, IAckableMessage, IMessageWithTracingId
+    {
+        /// <summary>
+        /// Gets or sets the ack id.
+        /// </summary>
+        public int AckId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tracing Id
+        /// </summary>
+        public ulong? TracingId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the list of excluded connection Ids.
+        /// </summary>
+        public IReadOnlyList<string> ExcludedList { get; set; }
+
+        /// <summary>
+        /// Gets or sets the reason closing the connection
+        /// </summary>
+        public string Reason { get; set; }
+
+        public CloseConnectionsWithAckMessage(int ackId)
```

Comment:
>int ackId [](http://example.com/codeflow?start=46&length=9)

why this ctor only contains ackId?

Reply:
because others are optional

Reply:
Only required parameters need to be in the ctor